src/boot/efi/stub.c

```diff
@@ -421,3 +416,15 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
         graphics_mode(false);
         return err;
 }
+
+EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
+        InitializeLib(image, sys_table);
+
+        debug_hook(L"systemd-stub");
+        /* Uncomment the next line if you need to wait for debugger. */
+        // debug_break();
+
+        EFI_STATUS err = real_main(image);
+        log_wait();
+        return err;
+}
```

Comment:
this reminds me a lot of what we have in userspace with DEFINE_MAIN_FUNCTION(). I think we might want to do something similar here sooner or later?

(but i guess that can be done another time)

Reply:
I'll think on it